tests/test_discovery.py

```diff
@@ -53,3 +54,27 @@ def test_user_only(tmp_path: Path, monkeypatch: pytest.MonkeyPatch, capfd: pytes
     expected = {"foo"}
 
     assert found == expected
+
+
+def test_duplicate_metadata(monkeypatch: pytest.MonkeyPatch, capfd: pytest.CaptureFixture[str]) -> None:
+    monkeypatch.setattr(
```

Comment:
Do not use monkey patch for anything other than setting or deleting environment variables. You should be using pytest-mock instead.

Reply:
Gotcha, I'll see what I can do

Reply:
The test code now uses `pytest_mock.MockerFixture`. Seeing there seems to be other areas where `monkeypatch` has been misused, I'll see if I can open a issue/PR in the future to change these areas to use pytest_mock instead.